Components/Games/NewGame.jsx

```diff
@@ -109,72 +134,61 @@ class NewGame extends React.Component {
         );
     }
 
+    getGameTypeOptions() {
+        return (
+            <div className='row'>
+                <div className='col-sm-12'>
+                    <b>Game Type</b>
+                </div>
+                <div className='col-sm-10'>
+                    <label className='radio-inline'>
+                        <input type='radio' onChange={ this.onRadioChange.bind(this, 'beginner') } checked={ this.isGameTypeSelected('beginner') } />
+                        Beginner
+                    </label>
+                    <label className='radio-inline'>
+                        <input type='radio' onChange={ this.onRadioChange.bind(this, 'casual') } checked={ this.isGameTypeSelected('casual') } />
+                        Casual
+                    </label>
+                    <label className='radio-inline'>
+                        <input type='radio' onChange={ this.onRadioChange.bind(this, 'competitive') } checked={ this.isGameTypeSelected('competitive') } />
+                        Competitive
+                    </label>
+                </div>
+            </div>);
+    }
+
     render() {
         let charsLeft = GameNameMaxLength - this.state.gameName.length;
         return this.props.socket ? (
             <div>
-                <Panel title='New game'>
+                <Panel title={ this.props.quickJoin ? 'Join Existing or Start New Game' : 'New game' }>
```

Comment:
There's enough of these checks for `quickJoin` that it feels like it would be better served to create a separate `QuickJoinGame` component. I can see the temptation to do it here due to the underlying shared functionality, though.

Your call. I might try to see if better components can be extracted later if merged as is.

Reply:
There's too much common functionality for it to be an entirely separate component, but some of the internal stuff could get factored out.  My thinking was to leave it until melee comes along as it'll need to be rethought a bit there anyway